pkg/virtctl/sev/attestation.go

```diff
@@ -0,0 +1,290 @@
+			c := SEVCommand{
+				command:      COMMAND_QUERY_MEASUREMENT,
+				clientConfig: clientConfig,
+			}
+			return c.Run(args)
+		},
+	}
+
+	cmd.SetUsageTemplate(templates.UsageTemplate())
+	cmd.Flags().StringVar(&outpath, outputFlag, "",
+		"Filepath where platform measurement info is to be written")
+	return cmd
+}
+
+func NewInjectSecretCommand(clientConfig clientcmd.ClientConfig) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:     "inject-secret (VMI)",
+		Short:   "Inject the guest owner's secret into the running VM",
+		Long:    `Inject the guest owner's secret into the running VM`,
+		Args:    templates.ExactArgs(COMMAND_INJECT_SECRET, 1),
+		Example: usage(COMMAND_INJECT_SECRET),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			c := SEVCommand{
+				command:      COMMAND_INJECT_SECRET,
+				clientConfig: clientConfig,
+			}
+			return c.Run(args)
+		},
+	}
+
+	cmd.SetUsageTemplate(templates.UsageTemplate())
+	cmd.Flags().StringVar(&secret, secretFlag, "", "(Required) Guest owner secret key")
+	cmd.MarkFlagRequired(secretFlag)
+	cmd.Flags().StringVar(&header, headerFlag, "", "(Required) Guest owner secret key header")
+	cmd.MarkFlagRequired(headerFlag)
+
+	return cmd
+}
+
+func usage(cmd string) string {
+	if cmd == COMMAND_SEV {
+		return fmt.Sprintf(" {{ProgramName}} sev <subcommand> myvmi")
+	} else {
+		return fmt.Sprintf(" {{ProgramName}} sev %s myvmi", cmd)
+	}
+}
+
+func (sc *SEVCommand) Run(args []string) error {
+	vmiName := args[0]
+	namespace, _, err := sc.clientConfig.Namespace()
+	if err != nil {
+		return err
+	}
+
+	virtClient, err := kubecli.GetKubevirtClientFromClientConfig(sc.clientConfig)
+	if err != nil {
+		return fmt.Errorf("Cannot obtain KubeVirt client: %v", err)
+	}
+
+	switch sc.command {
+	case COMMAND_SEV:
+		return fmt.Errorf("A subcommand is needed to execute the sev command")
+
+	case COMMAND_FETCH_CERT_CHAIN:
+		var file *os.File
+		pathProvided := outpath != ""
+		if pathProvided {
+			file, err = os.OpenFile(outpath, os.O_WRONLY|os.O_CREATE|os.O_EXCL, output_perm)
+			if err != nil {
+				log.Log.Errorf("[SEV] Failed to write certificate chain to %s. File already exists", outpath)
+				return fmt.Errorf("Error %s/%s: %v", namespace, vmiName, err)
+			}
+		}
+		defer file.Close()
+
+		if sevPlatformInfo, err := virtClient.VirtualMachineInstance(namespace).SEVFetchCertChain(vmiName); err != nil {
+			return fmt.Errorf("Error %s/%s: %v", namespace, vmiName, err)
+		} else {
+			data, err := json.MarshalIndent(sevPlatformInfo, "", "  ")
+			if err != nil {
+				return fmt.Errorf("Cannot marshal SEV platform info %v", err)
+			}
+			if pathProvided {
+				if _, err := file.Write(data); err != nil {
+					log.Log.Errorf("[SEV] Failed to write certificate chain to %s", outpath)
+					return fmt.Errorf("Error %s/%s: %v", namespace, vmiName, err)
+				}
+				log.Log.Infof("[SEV]Writing platform certificate chain to %s", outpath)
+			} else {
+				fmt.Printf("%s\n", string(data))
+			}
```

Comment:
I think the code can be simplified by using https://pkg.go.dev/io/ioutil#WriteFile . That way you will not need the `file, err = os.OpenFile(...)` part above. Same applies for the `COMMAND_QUERY_MEASUREMENT` case.

Reply:
Using `WriteFile()` would overwrite the file if it already existed. As designed, I don't allow overwriting existing files and it's a failure condition if the file exists.